File: p191e/hamming_weight.js
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function (n) {
  let ans = 0;
  while (n) {
    n &= n - 1;
    ans++;
  }
  return ans;
};
// TEST
[
  [0b00000000000000000000000000001011, 3],
  [0b00000000000000000000000010000000, 1],
  [0b11111111111111111111111111111101, 31],
].forEach(([n, expected]) => {
  const actual = hammingWeight(n);
  console.log('Hamming weight of', n.toString(2), '->', actual);
  console.assert(actual === expected);
});

File: p191e/hamming_weight.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        ans = 0
        while n:
            n &= n - 1
            ans += 1
        return ans


# TESTS
for n, expected in [
    (0b00000000000000000000000000001011, 3),
    (0b00000000000000000000000010000000, 1),
    (0b11111111111111111111111111111101, 31),
]:
    sol = Solution()
    actual = sol.hammingWeight(n)
    print("Hamming weight of", bin(n), "->", actual)
    assert actual == expected
